classes/Smarty/TemplateFinder.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/*
+* 2007-2015 PrestaShop
+*
+* NOTICE OF LICENSE
+*
+* This source file is subject to the Open Software License (OSL 3.0)
+* that is bundled with this package in the file LICENSE.txt.
+* It is also available through the world-wide-web at this URL:
+* http://opensource.org/licenses/osl-3.0.php
+* If you did not receive a copy of the license and are unable to
+* obtain it through the world-wide-web, please send an email
+* to license@prestashop.com so we can send you a copy immediately.
+*
+* DISCLAIMER
+*
+* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to http://www.prestashop.com for more information.
+*
+*  @author PrestaShop SA <contact@prestashop.com>
+*  @copyright  2007-2015 PrestaShop SA
+*  @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+*  International Registered Trademark & Property of PrestaShop SA
+*/
+
+/**
+ * Determine the best existing template.
+ *
+ * @since 1.7.0.0
+ */
+class TemplateFinderCore
+{
+    private $directories;
+    private $extension;
+    private $productListEntities = array('category', 'manufacturer', 'supplier');
+    private $productListSearchEntities = array('search', 'price-drop', 'best-sale');
+    private $productEntities = array('product');
+
+    public function __construct(array $directories, $extension)
+    {
+        $this->directories = $directories;
+        $this->extension = $extension;
+    }
+
+    public function getTemplate($template, $entity, $id, $locale)
+    {
+        $locale = (Validate::isLocale($locale)) ? $locale : '';
+
+        $templates = $this->getTemplateHierarchy($template, $entity, $id, $locale);
+
+        foreach ($this->directories as $dir) {
+            foreach ($templates as $tpl) {
+                if (file_exists($dir.$locale.DIRECTORY_SEPARATOR.$tpl.$this->extension)) {
```

Comment:
because this can be true if a folder is returned. Maybe ``is_file`` can be used in addition ?